sugar.go

```diff
@@ -213,6 +214,19 @@ func (p *Page) Screenshot(toFile ...string) []byte {
 	return bin
 }
 
+// FullScreenshot gets the full height of the page and returns the binary of the image.
+// Uses an internal JavaScript function to get the client height.
+func (p *Page) FullScreenshot(toFile ...string) []byte {
+	height := p.Eval(`() => document.body.clientHeight`).String()
+	heightToInt, err := strconv.ParseInt(height, 10, 64)
```

Comment:
Check the doc of [window size](https://developer.mozilla.org/en-US/docs/Web/API/Window/innerWidth)

```go
size := p.Eval(`() => {w: innerWidth, h: innerHeight}`)
kit.Log(size.Get("w").Int(), size.Get("h").Int())
```

Reply:
I did it a bit differently - added a flag to `fullScreenshot` to switch between HD (1920) and the viewport because the `document.body.clientWidth` or `window.innerWidth` will always return the width of the window.

Maybe we could make the `fullScreenshot` enforce the device emulation? So then we'd have a function to do a full screenshot of a page as viewed on a phone or on a desktop?

Reply:
Sorry my bad, we shouldn't use clientHeight or clientWidth at all.
Here's the doc to check: https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-getLayoutMetrics

The `contentSize` is the full page size including the scroll size.

Reply:
You might want to check this example to call cdp low-level API:

https://github.com/ysmood/rod/blob/9b20ec0ccc4293eed0af0a8b659ee085790d1fa9/examples_test.go#L181-L189

The IDE auto-complete will guide you the correct name of the `getLayoutMetrics`

Reply:
Ok, let's roll with it!